esmvaltool/cmorizers/data/formatters/datasets/esacci_cloud.py

```diff
@@ -0,0 +1,404 @@
+    ):
+        cubes_pm.append(monthly_cube)
+    else:
+        logger.error("The file %s is not assigned to AM or PM", ifile)
+
+
+def _extract_variable_daily(
+    short_name, var, cfg, in_dir, out_dir, start_date, end_date
+):
+    """Extract daily variable."""
+    cmor_info = cfg["cmor_table"].get_variable(var["mip"], short_name)
+
+    if not start_date:
+        start_date = datetime(cfg["start_year_daily"], 1, 1)
+    if not end_date:
+        end_date = datetime(cfg["end_year_daily"], 12, 31)
+
+    for year in range(start_date.year, end_date.year + 1):
+        cubes = iris.cube.CubeList()
+        cubes_day = iris.cube.CubeList()
+
+        for month in range(1, 13):
+            num_days = monthrange(year, month)[1]
+            for iday in range(1, num_days + 1):
+                filelist = glob.glob(
+                    os.path.join(
+                        in_dir,
+                        f"{year}{month:02}" + f"{iday:02}" + var["file"],
+                    )
+                )
+
+                if filelist:
+                    for inum, ifile in enumerate(filelist):
+                        try:
+                            _process_daily_file(
+                                ifile,
+                                inum,
+                                short_name,
+                                var,
+                                cmor_info,
+                                cubes,
+                                cubes_day,
+                            )
+                        except Exception as e:
+                            logger.error(
+                                "Error processing file '%s': %s", ifile, e
+                            )
+                else:
+                    _handle_missing_day(
+                        year, month, iday, short_name, cubes, cubes_day
+                    )
+
+        _concatenate_and_save_daily_cubes(
+            short_name, var, cubes, cubes_day, out_dir, cfg, cmor_info
+        )
+
+
+def _extract_variable_monthly(
+    short_name, var, cfg, in_dir, out_dir, start_date, end_date
+):
+    """Extract monthly variable with improved handling for multiple cubes."""
+    cmor_info = cfg["cmor_table"].get_variable(var["mip"], short_name)
+
+    if not start_date:
+        start_date = datetime(cfg["start_year_monthly"], 1, 1)
+    if not end_date:
+        end_date = datetime(cfg["end_year_monthly"], 12, 31)
+
+    cubes_am = iris.cube.CubeList()
+    cubes_pm = iris.cube.CubeList()
+
+    for year in range(start_date.year, end_date.year + 1):
+        for month in range(1, 13):  # Loop through all months (1-12)
+            # Construct the file list for the current month
+            filelist = glob.glob(
+                os.path.join(in_dir, f"{year}{month:02}" + var["file"])
+            )
+
+            if not filelist:
+                logger.warning(
+                    "No monthly file found for %s-%02d", year, month
+                )
+                continue
```

Comment:
Do you expect this to happen? If not, raise exception.

Reply:
Done.